File: basic_scala/src/test/scala/pers/xiaoming/spark/SparkCoreDemoTestBase.scala
package pers.xiaoming.spark

import org.apache.spark.{SparkConf, SparkContext}
import org.junit.{AfterClass, BeforeClass}

class SparkCoreDemoTestBase {

}

object SparkCoreDemoTestBase {
  var sc:SparkContext = _

  @BeforeClass
  def setup: Unit = {
    val conf = new SparkConf().setAppName("BasicDemo").setMaster("local")
    sc = new SparkContext(conf)
  }

  @AfterClass
  def close: Unit = {
    sc.stop()
  }
}

File: basic_java/src/test/java/pers/xiaoming/spark/SparkCoreDemoTestBase.java
package pers.xiaoming.spark;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaSparkContext;
import org.junit.AfterClass;
import org.junit.BeforeClass;

public class SparkCoreDemoTestBase {
    protected static JavaSparkContext sc;

    @BeforeClass
    public static void setup() {
        SparkConf config = new SparkConf().setAppName("BasicDemo").setMaster("local");
        sc = new JavaSparkContext(config);
    }

    @AfterClass
    public static void close() {
        if (sc != null) {
            sc.close();
        }
    }
}
